python/orca/src/bigdl/orca/data/shard.py

```diff
@@ -598,21 +598,32 @@ def utility_func(x, func, *args, **kwargs):
     def get_schema(self):
         if 'schema' in self.type:
             return self.type['schema']
-        else:
-            if self._get_class_name() == 'pandas.core.frame.DataFrame':
-                import pandas as pd
-                columns, dtypes = self.rdd.map(lambda x: (x.columns, x.dtypes)).first()
-                self.type['schema'] = {'columns': columns, 'dtypes': dtypes}
-                return self.type['schema']
-            return None
+
+        if 'class_name' not in self.type\
+                or self.type['class_name'] == 'pandas.core.frame.DataFrame':
+            class_name, schema = self._get_schema_class_name()
+            self.type['class_name'] = class_name
+            self.type['schema'] = schema
+            return self.type['schema']
```

Comment:
Need to return None otherwise

Reply:
Fixed in https://github.com/intel-analytics/BigDL/pull/5558